www/README.md

```diff
@@ -0,0 +1,121 @@
+* All ```generated-config.js``` is compiled into ```app.js```
+* Production config can be generated using ```gulp config --env production``` or ```gulp --env production```
+* The generated angular constant is named ```ENV``` with the shared root and each component having their own child configuration. For example,
+```
+www/master
+├── shared/config/development.json
+└── components
+    ├── dashboard/config/development.json
+    ├── graph/config/development.json
+    └── my_component/config/development.json
+```
+produces ```www/master/shared/config/generated-config.js```:
+```
+angular.module('kubernetesApp.config', [])
+.constant('ENV', {
+  '/': <www/master/shared/config/development.json>,
+  'dashboard': <www/master/components/dashboard/config/development.json>,
+  'graph': <www/master/components/graph/config/development.json>,
+  'my_component': <www/master/components/my_component/config/development.json>
+});
+```
+
+#### Kubernetes server configuration
+
+**RECOMMENDED**: By default the Kubernetes api server does not support CORS,
+  so the `kube-apiserver.service` must be started with
+  `--cors_allowed_origins=.*` or `--cors_allowed_origins=http://<your
+  host here>`
+
+**HACKS**: If you don't want to/cannot restart the Kubernetes api server:
+* Or you can start your browser with web security disabled. For
+  Chrome, you can [launch](http://www.chromium.org/developers/how-tos/run-chromium-with-flags) it with flag ```--disable-web-security```.
+
+### Building a new visualizer or component
+
+See [master/components/README.md](master/components/README.md).
+
+### Testing
+Currently kuberntes-ui includes both unit-testing (run via [Karma](http://karma-runner.github.io/0.12/index.html)) and
+end-to-end testing (run via
+[Protractor](http://angular.github.io/protractor/#/)).
+
+#### Unittests via Karma
+To run the existing Karma tests:
+* Install the Karma CLI: `sudo npm install -g karma-cli` (it needs to
+  be installed globally, hence the `sudo` may be needed). Note that
+  the other Karma packages (such as `karma`, `karma-jasmine`, and
+  `karma-chrome-launcher` should be automatically installed when
+  running `npm start`).
+* Go to the `www/master` directory, and run `karma start
+karma.conf.js`. The Karma configuration is defined in `karma.config.js`. The console should show the test results.
+
+To write new Karma tests:
+* For testing each components, write test files (`*.spec.js`) under the
+corresponding `www/master/components/**/test/modules/` directory.
+* For testing the chrome and the framework, write test files
+  (*.spec.js) under the `www/master/test/modules/*` directory.
+
+#### End-to-end testing via Protractor
+To run the existing Protractor tests:
+* Install the CLIs: `sudo npm install -g protractor`.
+* Start the webdriver server: `sudo webdriver-manager start`
```

Comment:
@lavalamp Unittests via Karma

To run the existing Karma tests:

Install the Karma CLI: sudo npm install -g karma-cli (it needs to be installed globally, hence the sudo may be needed). Note that the other Karma packages (such as karma, karma-jasmine, and karma-chrome-launcher should be automatically installed when running npm start).
Go to the www/master directory, and run karma start karma.conf.js. The Karma configuration is defined in karma.config.js. The console should show the test results.